costmap_2d/include/costmap_2d/layered_costmap.h

```diff
@@ -119,12 +123,17 @@ class LayeredCostmap
     return size_locked_;
   }
 
-  void getBounds(unsigned int* x0, unsigned int* xn, unsigned int* y0, unsigned int* yn)
+  void getBounds(unsigned int* x0, unsigned int* xn, unsigned int* y0, unsigned int* yn, double t = 0.0)
   {
-    *x0 = bx0_;
-    *xn = bxn_;
-    *y0 = by0_;
-    *yn = byn_;
+    int n;
+    if (!timestep_)
+      n = 0;
+    else
+      n = std::min((int)timed_bounds_.size()-1, int(t/timestep_));
```

Comment:
```suggestion
    int n = timestep == 0 ? 0 : std::min((int)timed_bounds_.size()-1, int(t/timestep_));
```
F: don't do boolean checks for numbers
S: use ternary operator
S: don't use `n` -- rename to something else like `i` since you're basically computing an index. `n` is usually used for amounts (e.g. compute the first N digits of PI, etc)